lib/irgen.ml

```diff
@@ -164,10 +191,11 @@ let translate (binds, sfuncs): L.llmodule =
           let pred_builder = L.builder_at_end context prd_bb in
           let bool_val = build_expr pred_builder prd in
       
-          let merge_bb = L.append_block context "merge" the_func in
+          let merge_bb = L.append_block context "merge_loop" the_func in
           ignore(L.build_cond_br bool_val block_bb merge_bb pred_builder);
           L.builder_at_end context merge_bb
       | SReturn sexp -> 
+          List.iter (fun arr -> ignore (L.build_free arr builder)) !malloced;
```

Comment:
simply commenting out this line fixes the bug. i guess we'll have to leave the `free`-ing up to the OS